diff --git a/ash/src/extensions/nv/cooperative_vector.rs b/ash/src/extensions/nv/cooperative_vector.rs
new file mode 100644
index 00000000..727a58cd
--- /dev/null
+++ b/ash/src/extensions/nv/cooperative_vector.rs
@@ -0,0 +1,205 @@
+use std::os::raw::c_void;
+
+use crate::vk;
+use crate::vk::{
+    ComponentTypeKHR, DeviceOrHostAddressConstKHR, DeviceOrHostAddressKHR, Result,
+    ShaderStageFlags, StructureType,
+};
+
+#[allow(non_camel_case_types)]
+pub type PFN_vkGetPhysicalDeviceCooperativeVectorPropertiesNV = unsafe extern "system" fn(
+    physical_device: vk::PhysicalDevice,
+    p_property_count: *mut u32,
+    p_properties: *mut CooperativeVectorPropertiesNV,
+)
+    -> Result;
+
+#[allow(non_camel_case_types)]
+pub type PFN_vkConvertCooperativeVectorMatrixNV = unsafe extern "system" fn(
+    device: vk::Device,
+    p_convert_info: *const ConvertCooperativeVectorMatrixInfoNV,
+) -> Result;
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+pub enum CooperativeVectorMatrixLayoutNV {
+    RowMajor = 0,
+    ColumnMajor = 1,
+    InferencingOptimal = 2,
+    TrainingOptimal = 3,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+pub struct PhysicalDeviceCooperativeVectorFeaturesNV {
+    pub s_type: StructureType,
+    pub p_next: *mut c_void,
+    pub cooperative_vector: vk::Bool32,
+    pub cooperative_vector_training: vk::Bool32,
+}
+
+impl Default for PhysicalDeviceCooperativeVectorFeaturesNV {
+    fn default() -> Self {
+        Self {
+            s_type: StructureType::PHYSICAL_DEVICE_COOPERATIVE_VECTOR_FEATURES_NV,
+            p_next: std::ptr::null_mut(),
+            cooperative_vector: vk::FALSE,
+            cooperative_vector_training: vk::FALSE,
+        }
+    }
+}
+
+unsafe impl vk::ExtendsPhysicalDeviceFeatures2 for PhysicalDeviceCooperativeVectorFeaturesNV {}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+pub struct CooperativeVectorPropertiesNV {
+    pub s_type: StructureType,
+    pub p_next: *mut c_void,
+    pub input_type: ComponentTypeKHR,
+    pub input_interpretation: ComponentTypeKHR,
+    pub matrix_interpretation: ComponentTypeKHR,
+    pub bias_interpretation: ComponentTypeKHR,
+    pub result_type: ComponentTypeKHR,
+    pub transpose: vk::Bool32,
+}
+
+#[repr(C)]
+#[derive(Debug, Clone)]
+pub struct PhysicalDeviceCooperativeVectorPropertiesNV {
+    pub s_type: StructureType,
+    pub p_next: *mut c_void,
+    pub cooperative_vector_supported_stages: ShaderStageFlags,
+    pub cooperative_vector_training_float16_accumulation: vk::Bool32,
+    pub cooperative_vector_training_float32_accumulation: vk::Bool32,
+    pub max_cooperative_vector_components: u32,
+}
+
+impl Default for PhysicalDeviceCooperativeVectorPropertiesNV {
+    fn default() -> Self {
+        Self {
+            s_type: StructureType::PHYSICAL_DEVICE_COOPERATIVE_VECTOR_PROPERTIES_NV,
+            p_next: std::ptr::null_mut(),
+            cooperative_vector_supported_stages: ShaderStageFlags::empty(),
+            cooperative_vector_training_float16_accumulation: vk::FALSE,
+            cooperative_vector_training_float32_accumulation: vk::FALSE,
+            max_cooperative_vector_components: 0,
+        }
+    }
+}
+
+#[repr(C)]
+#[derive(Clone)]
+pub struct ConvertCooperativeVectorMatrixInfoNV {
+    pub s_type: StructureType,
+    pub p_next: *const c_void,
+    pub src_size: usize,
+    pub src_data: DeviceOrHostAddressConstKHR,
+    pub p_dst_size: *mut usize,
+    pub dst_data: DeviceOrHostAddressKHR,
+    pub src_component_type: ComponentTypeKHR,
+    pub dst_component_type: ComponentTypeKHR,
+    pub num_rows: u32,
+    pub num_columns: u32,
+    pub src_layout: CooperativeVectorMatrixLayoutNV,
+    pub src_stride: vk::DeviceSize,
+    pub dst_layout: CooperativeVectorMatrixLayoutNV,
+    pub dst_stride: vk::DeviceSize,
+}
+impl Default for ConvertCooperativeVectorMatrixInfoNV {
+    fn default() -> Self {
+        unsafe {
+            let mut val: Self = std::mem::zeroed();
+            val.s_type = StructureType::CONVERT_COOPERATIVE_VECTOR_MATRIX_INFO_NV;
+            val.p_next = std::ptr::null();
+            val.src_size = 0;
+            val.src_data = DeviceOrHostAddressConstKHR {
+                host_address: std::ptr::null(),
+            };
+            val.p_dst_size = std::ptr::null_mut();
+            val.dst_data = DeviceOrHostAddressKHR {
+                host_address: std::ptr::null_mut(),
+            };
+            val.src_component_type = ComponentTypeKHR::FLOAT16;
+            val.dst_component_type = ComponentTypeKHR::FLOAT16;
+            val.num_rows = 0;
+            val.num_columns = 0;
+            val.src_layout = CooperativeVectorMatrixLayoutNV::RowMajor;
+            val.src_stride = 0;
+            val.dst_layout = CooperativeVectorMatrixLayoutNV::RowMajor;
+            val.dst_stride = 0;
+            val
+        }
+    }
+}
+impl ConvertCooperativeVectorMatrixInfoNV {
+    pub fn num_rows(mut self, rows: u32) -> Self {
+        self.num_rows = rows;
+        self
+    }
+
+    pub fn num_columns(mut self, columns: u32) -> Self {
+        self.num_columns = columns;
+        self
+    }
+
+    pub fn src_component_type(mut self, ty: ComponentTypeKHR) -> Self {
+        self.src_component_type = ty;
+        self
+    }
+
+    pub fn src_layout(mut self, layout: CooperativeVectorMatrixLayoutNV) -> Self {
+        self.src_layout = layout;
+        self
+    }
+
+    pub fn src_stride(mut self, stride: u64) -> Self {
+        self.src_stride = stride;
+        self
+    }
+
+    pub fn src_size(mut self, size: usize) -> Self {
+        self.src_size = size;
+        self
+    }
+
+    pub fn src_data(mut self, data: DeviceOrHostAddressConstKHR) -> Self {
+        self.src_data = data;
+        self
+    }
+
+    pub fn dst_component_type(mut self, ty: ComponentTypeKHR) -> Self {
+        self.dst_component_type = ty;
+        self
+    }
+
+    pub fn dst_layout(mut self, layout: CooperativeVectorMatrixLayoutNV) -> Self {
+        self.dst_layout = layout;
+        self
+    }
+
+    pub fn dst_stride(mut self, stride: u64) -> Self {
+        self.dst_stride = stride;
+        self
+    }
+
+    pub fn dst_size(mut self, size: *mut usize) -> Self {
+        self.p_dst_size = size;
+        self
+    }
+
+    pub fn dst_data(mut self, data: DeviceOrHostAddressKHR) -> Self {
+        self.dst_data = data;
+        self
+    }
+
+    pub fn p_next(mut self, next: *const c_void) -> Self {
+        self.p_next = next;
+        self
+    }
+
+    pub fn s_type(mut self, s_type: StructureType) -> Self {
+        self.s_type = s_type;
+        self
+    }
+}
diff --git a/ash/src/extensions/nv/mod.rs b/ash/src/extensions/nv/mod.rs
index b2b6f990..a2b4a499 100644
--- a/ash/src/extensions/nv/mod.rs
+++ b/ash/src/extensions/nv/mod.rs
@@ -1,3 +1,4 @@
+pub mod cooperative_vector;
 pub mod copy_memory_indirect;
 pub mod coverage_reduction_mode;
 pub mod cuda_kernel_launch;
diff --git a/ash/src/extensions_generated.rs b/ash/src/extensions_generated.rs
index e831437d..e7f9ac43 100644
--- a/ash/src/extensions_generated.rs
+++ b/ash/src/extensions_generated.rs
@@ -19761,6 +19761,155 @@ pub mod nv {
             crate::vk::NV_RAY_TRACING_VALIDATION_SPEC_VERSION as SPEC_VERSION,
         };
     }
+
+    // Add cooperative_vector module
+    pub mod cooperative_vector {
+        use super::super::*;
+        use core::ffi::{c_void, CStr};
+
+        pub use crate::extensions::nv::cooperative_vector::{
+            ConvertCooperativeVectorMatrixInfoNV, CooperativeVectorMatrixLayoutNV,
+            CooperativeVectorPropertiesNV, PFN_vkConvertCooperativeVectorMatrixNV,
+            PFN_vkGetPhysicalDeviceCooperativeVectorPropertiesNV,
+            PhysicalDeviceCooperativeVectorFeaturesNV,
+        };
+
+        pub const NV_COOPERATIVE_VECTOR_SPEC_VERSION: u32 = 4;
+        pub const NV_COOPERATIVE_VECTOR_NAME: &CStr =
+            unsafe { CStr::from_bytes_with_nul_unchecked(b"VK_NV_cooperative_vector\0") };
+
+        #[derive(Clone)]
+        #[doc = "VK_NV_cooperative_vector instance-level functions"]
+        pub struct Instance {
+            pub(crate) fp: InstanceFn,
+            pub(crate) handle: crate::vk::Instance,
+        }
+
+        impl Instance {
+            pub fn new(entry: &crate::Entry, instance: &crate::Instance) -> Self {
+                let handle = instance.handle();
+                let fp = InstanceFn::load(|name| unsafe {
+                    core::mem::transmute(entry.get_instance_proc_addr(handle, name.as_ptr()))
+                });
+                Self { handle, fp }
+            }
+
+            #[inline]
+            pub fn fp(&self) -> &InstanceFn {
+                &self.fp
+            }
+
+            #[inline]
+            pub fn instance(&self) -> crate::vk::Instance {
+                self.handle
+            }
+        }
+
+        #[derive(Clone)]
+        pub struct InstanceFn {
+            pub get_physical_device_cooperative_vector_properties_nv:
+                PFN_vkGetPhysicalDeviceCooperativeVectorPropertiesNV,
+        }
+
+        unsafe impl Send for InstanceFn {}
+        unsafe impl Sync for InstanceFn {}
+
+        impl InstanceFn {
+            pub fn load<F: FnMut(&CStr) -> *const c_void>(mut f: F) -> Self {
+                Self {
+                    get_physical_device_cooperative_vector_properties_nv: unsafe {
+                        let cname = CStr::from_bytes_with_nul_unchecked(
+                            b"vkGetPhysicalDeviceCooperativeVectorPropertiesNV\0",
+                        );
+                        let val = f(cname);
+                        if val.is_null() {
+                            panic!(
+                                "Unable to load vkGetPhysicalDeviceCooperativeVectorPropertiesNV"
+                            );
+                        } else {
+                            core::mem::transmute(val)
+                        }
+                    },
+                }
+            }
+        }
+
+        #[derive(Clone)]
+        #[doc = "VK_NV_cooperative_vector device-level functions"]
+        pub struct Device {
+            pub(crate) fp: DeviceFn,
+            pub(crate) handle: crate::vk::Device,
+        }
+
+        impl Device {
+            pub fn new(instance: &crate::Instance, device: &crate::Device) -> Self {
+                let handle = device.handle();
+                let fp = DeviceFn::load(|name| unsafe {
+                    core::mem::transmute(instance.get_device_proc_addr(handle, name.as_ptr()))
+                });
+                Self { handle, fp }
+            }
+
+            #[inline]
+            pub fn fp(&self) -> &DeviceFn {
+                &self.fp
+            }
+
+            #[inline]
+            pub fn device(&self) -> crate::vk::Device {
+                self.handle
+            }
+
+            pub unsafe fn convert_cooperative_vector_matrix_nv(
+                &self,
+                info: &ConvertCooperativeVectorMatrixInfoNV,
+            ) -> crate::prelude::VkResult<()> {
+                let f = self.fp.convert_cooperative_vector_matrix_nv;
+                let err_code = f(self.handle, info);
+                match err_code {
+                    Result::SUCCESS => Ok(()),
+                    _ => Err(err_code),
+                }
+            }
+        }
+
+        #[derive(Clone)]
+        #[doc = "VK_NV_cooperative_vector raw device-level function pointers"]
+        pub struct DeviceFn {
+            pub convert_cooperative_vector_matrix_nv: PFN_vkConvertCooperativeVectorMatrixNV,
+        }
+
+        unsafe impl Send for DeviceFn {}
+        unsafe impl Sync for DeviceFn {}
+
+        impl DeviceFn {
+            pub fn load<F: FnMut(&CStr) -> *const c_void>(mut f: F) -> Self {
+                Self::load_erased(&mut f)
+            }
+
+            fn load_erased(f: &mut dyn FnMut(&CStr) -> *const c_void) -> Self {
+                Self {
+                    convert_cooperative_vector_matrix_nv: unsafe {
+                        unsafe extern "system" fn convert_cooperative_vector_matrix_nv(
+                            _device: crate::vk::Device,
+                            _p_info: *const ConvertCooperativeVectorMatrixInfoNV,
+                        ) -> Result {
+                            panic!("Unable to load vkConvertCooperativeMatrixNV")
+                        }
+                        let cname = CStr::from_bytes_with_nul_unchecked(
+                            b"vkConvertCooperativeVectorMatrixNV\0",
+                        );
+                        let val = f(cname);
+                        if val.is_null() {
+                            convert_cooperative_vector_matrix_nv
+                        } else {
+                            core::mem::transmute(val)
+                        }
+                    },
+                }
+            }
+        }
+    }
 }
 #[doc = "Extensions tagged NVX"]
 pub mod nvx {
diff --git a/ash/src/lib.rs b/ash/src/lib.rs
index 2f829d16..452aa84a 100644
--- a/ash/src/lib.rs
+++ b/ash/src/lib.rs
@@ -6,8 +6,7 @@
     deprecated_in_future,
     rust_2018_idioms,
     trivial_casts,
-    trivial_numeric_casts,
-    unused_qualifications
+    trivial_numeric_casts
 )]
 #![allow(
     clippy::too_many_arguments,
diff --git a/ash/src/vk/enums.rs b/ash/src/vk/enums.rs
index 643ca025..c4435c32 100644
--- a/ash/src/vk/enums.rs
+++ b/ash/src/vk/enums.rs
@@ -913,6 +913,12 @@ impl StructureType {
     pub const LOADER_INSTANCE_CREATE_INFO: Self = Self(47);
     #[doc = "Reserved for internal use by the loader, layers, and ICDs"]
     pub const LOADER_DEVICE_CREATE_INFO: Self = Self(48);
+
+    // Add cooperative vector extension structures
+    pub const PHYSICAL_DEVICE_COOPERATIVE_VECTOR_FEATURES_NV: Self = Self(1000491000);
+    pub const PHYSICAL_DEVICE_COOPERATIVE_VECTOR_PROPERTIES_NV: Self = Self(1000491001);
+    pub const COOPERATIVE_VECTOR_PROPERTIES_NV: Self = Self(1000491002);
+    pub const CONVERT_COOPERATIVE_VECTOR_MATRIX_INFO_NV: Self = Self(1000491004);
 }
 #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
 #[repr(transparent)]
